atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/CachingTransaction.java

```diff
@@ -182,8 +186,51 @@ private void addToColCache(TableReference tableRef, Map<Cell, byte[]> values) {
         }
     }
 
-    private ConcurrentMap<Cell, byte[]> getColCacheForTable(TableReference tableRef) {
+    private LoadingCache<Cell, byte[]> getColCacheForTable(TableReference tableRef) {
         String tableName = tableRef.getQualifiedName();
-        return columnTableCache.getUnchecked(tableName);
+        try {
+            return columnTableCache.get(tableName, () -> createNewCacheForTable(tableRef));
+        } catch (ExecutionException e) {
+            throw Throwables.throwUncheckedException(e.getCause());
+        }
+    }
+
+    private LoadingCache<Cell, byte[]> createNewCacheForTable(TableReference tableRef) {
+        return CacheBuilder.newBuilder()
+                .softValues()
+                .maximumWeight(maxCacheWeightPerTable)
+                .weigher((Weigher<Cell, byte[]>) (key, value) -> value.length)
+                .build(getCacheLoader(tableRef));
```

Comment:
Should we log eviction at debug or info so that folks are aware of insufficient size if they are concerned about reading cached values?